include/deal.II/trilinos/nox.h

```diff
@@ -0,0 +1,1029 @@
+      {}
+
+      NOX::StatusTest::StatusType
+      checkStatus(const NOX::Solver::Generic &problem,
+                  NOX::StatusTest::CheckType  checkType) override
+      {
+        if (checkType == NOX::StatusTest::None)
+          {
+            status = NOX::StatusTest::Unevaluated;
+          }
+        else
+          {
+            if (check_iteration_status == nullptr)
+              {
+                status = NOX::StatusTest::Converged;
+              }
+            else
+              {
+                const auto &x = problem.getSolutionGroup().getX();
+                const auto *x_ =
+                  dynamic_cast<const internal::Vector<VectorType> *>(&x);
+
+                const auto &f = problem.getSolutionGroup().getF();
+                const auto *f_ =
+                  dynamic_cast<const internal::Vector<VectorType> *>(&f);
+
+                const unsigned int step = problem.getNumIterations();
+
+                const double norm_f = f_->genericVector().l2_norm();
+
+                state = this->check_iteration_status(step,
+                                                     norm_f,
+                                                     x_->genericVector(),
+                                                     f_->genericVector());
+
+                switch (state)
+                  {
+                    case SolverControl::iterate:
+                      status = NOX::StatusTest::Unconverged;
+                      break;
+                    case SolverControl::failure:
+                      status = NOX::StatusTest::Failed;
+                      break;
+                    case SolverControl::success:
+                      status = NOX::StatusTest::Converged;
+                      break;
+                    default:
+                      AssertThrow(false, ExcNotImplemented());
+                  }
+              }
+          }
+
+        if (as_dummy)
+          status = NOX::StatusTest::Unconverged;
+
+        return status;
+      }
+
+      NOX::StatusTest::StatusType
+      getStatus() const override
+      {
+        return status;
+      }
+
+      virtual std::ostream &
+      print(std::ostream &stream, int indent = 0) const override
+      {
+        (void)indent;
+
+        std::string state_str;
+        switch (state)
+          {
+            case SolverControl::iterate:
+              state_str = "iterate";
+              break;
+            case SolverControl::failure:
+              state_str = "failure";
+              break;
+            case SolverControl::success:
+              state_str = "success";
+              break;
+            default:
+              AssertThrow(false, ExcNotImplemented());
+          }
+
+        for (int j = 0; j < indent; j++)
+          stream << ' ';
+        stream << status;
+        stream << "check_iteration_status() = " << state_str
+               << " (dummy = " << (as_dummy ? "yes" : "no") << ")";
+        stream << std::endl;
```

Comment:
If you move the first three lines (the whitespace padding + outputting the status code) above the `switch` block, then you can directly `stream << "iterate";` in the various `case` cases, without the need for the `state_str` temp variable.

Separately, though, did you want a space or other character between the `status` and `"check_iteration_status()"`?